sigstore/_utils.py

```diff
@@ -32,8 +32,15 @@
 else:
     from importlib import resources
 
+
 PublicKey = Union[rsa.RSAPublicKey, ec.EllipticCurvePublicKey]
 
+hexstr = NewType("hexstr", str)
```

Comment:
These won't be public APIs, but let's add docstrings for them anyways, e.g.:

```suggestion
hexstr = NewType("hexstr", str)
"""
A newtype for `str` objects that contain hexadecimal strings (e.g. `ffabcd00ff`).
"""
```